runelite-client/src/main/java/net/runelite/client/plugins/clanchat/ClanChatPlugin.java

```diff
@@ -456,6 +469,20 @@ public void onChatMessage(ChatMessage chatMessage)
 				}
 				break;
 			case FRIENDSCHAT:
+				if (config.senderLimit())
+				{
+					chatMessage.getMessageNode()
+						.setSender(chatMessage.getMessageNode()
+							.getSender().substring(0, config.senderLength()));
+				}
+				// Name replacement overrides character limit
+				if (!config.senderReplace().isEmpty())
+				{
+					chatMessage.getMessageNode()
+						.setSender(config.senderReplace());
+				}
+				client.refreshChat();
```

Comment:
If neither of the conditions are hit, there would not be any need to refresh chat. I'd like to check for that also instead of refreshing on all FRIENDSCHAT messages no matter what.